### PR TITLE
132 subschema ihc

### LINKED ISSUE #132: Subschema IHC
Subschema for IHC

IHC - Pathreports
IHC - biospecimen

### PR DESCRIPTION
**Pull request by SenseCode**
close #132

### CODE REVIEW

File: src/encoded/schemas/pathology_report.json [hunk 1]
```diff
@@ -0,0 +1,604 @@
...
+        "description": "AJCC TNM Stage",
+        "type": "string",
+        "enum": [
+                "1",
+                "2",
+                "3",
+                "4"
+            ]
+      },
+      "ajcc_p_stage": {
```
Comment (mingjiecn) [inline src/encoded/schemas/pathology_report.json:428]: "ajcc_p_stage"( "title": "AJCC pT Stage") seems to be a duplicated property.

File: src/encoded/static/components/patholgoyReprotTable.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                                </div>
+                                <div data-test="histology">
+                                    <dt>Histology</dt>
+                                    <dd>{this.filteredData[i].histology}</dd>
+                                </div>
+                                <div data-test="sarcomatoid">
+                                    <dt>Sarcomatoid</dt>
+                                    <dd>{this.filteredData[i].sarcomatoid}</dd>
+                                </div>
+                                <div data-test="sarcomatoid_percentage">
```
Comment (mingjiecn) [inline src/encoded/static/components/patholgoyReprotTable.js:55]: Hide the field sarcomatoid_percentage if there is no value for this field

File: src/encoded/static/components/pathology_report.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+                            <div data-test="histology">
+                                <dt>Histology</dt>
+                                <dd>{context.histology}</dd>
+                            </div>
+                            <div data-test="sarcomatoid">
+                                <dt>Sarcomatoid</dt>
+                                <dd>{context.sarcomatoid}</dd>
+                            </div>
+                            <div data-test="sarcomatoid_percentage">
+                                <dt>Sarcomatoid_percentage</dt>
```
Comment (mingjiecn) [inline src/encoded/static/components/pathology_report.js:75]: Hide the field sarcomatoid_percentage if there is no value for this field

File: src/encoded/static/components/surgery.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                        </div>
+                        <div data-test="surgery date">
+                            <dt>Surgery date</dt>
+                            <dd>{context.date}</dd>
+                        </div>
+                        <div data-test="hospital">
+                            <dt>Hospital Location</dt>
+                            <dd>{context.hospital_location}</dd>
+                        </div>
+
```
Comment (mingjiecn) [inline src/encoded/static/components/surgery.js:72]: Should have more info for surgery?

File: src/encoded/schemas/pathology_report.json (hunk 1 above)
Comment (SenseCode) [reply src/encoded/schemas/pathology_report.json:428]: Not matters currently I think.

File: src/encoded/static/components/patholgoyReprotTable.js (hunk 2 above)
Comment (SenseCode) [reply src/encoded/static/components/patholgoyReprotTable.js:55]: We can filter all the show fields after we decide which fields need to be shown. Agree to fix this.

File: src/encoded/static/components/pathology_report.js (hunk 3 above)
Comment (SenseCode) [reply src/encoded/static/components/pathology_report.js:75]: Same fix.

File: src/encoded/static/components/surgery.js (hunk 4 above)
Comment (SenseCode) [reply src/encoded/static/components/surgery.js:72]: Currently we didn't show the surgery procedure detail. Some surgery have more types here, might no enough space to show all the information.

image

Comment (vsmalladi) [reply src/encoded/static/components/surgery.js:72]: Could we do a list?

File: src/encoded/schemas/pathology_report.json (hunk 1 above)
Comment (vsmalladi) [reply src/encoded/schemas/pathology_report.json:428]: We should go over all and see what are redundant

**Review by vsmalladi — APPROVED** [review]

**Pull request merged**